test/parallel/test-webstreams-pipeline.js

```diff
@@ -0,0 +1,392 @@
+  }));
+
+  c.enqueue('hello');
+  c.enqueue('world');
+  c.close();
+}
+
+{
+  let c;
+  const rs = new ReadableStream({
+    start(controller) {
+      c = controller;
+    }
+  });
+
+  const ws = new WritableStream({
+    write() { }
+  });
+
+  pipeline(rs, ws, common.mustCall((err) => {
+    assert.strictEqual(err?.message, 'kaboom');
+  }));
+
+  c.error(new Error('kaboom'));
+}
+
+{
+  let c;
+  const values = [];
+  const rs = new ReadableStream({
+    start(controller) {
+      c = controller;
+    }
+  });
+
+  const ts = new TransformStream({
+    transform(chunk, controller) {
+      controller.enqueue(chunk?.toString().toUpperCase());
+    }
+  });
+
+  const ws = new WritableStream({
+    write(chunk) {
+      values.push(chunk?.toString());
+    }
+  });
+
+  pipeline(rs, ts, ws, common.mustSucceed(() => {
+    assert.deepStrictEqual(values, ['HELLO', 'WORLD']);
+  }));
+
+  c.enqueue('hello');
+  c.enqueue('world');
+  c.close();
+}
+
+{
+  function makeTransformStream() {
+    return new TransformStream({
+      transform(chunk, controller) {
+        controller.enqueue(chunk?.toString());
+      }
+    });
+  }
+
+  let c;
+  const rs = new ReadableStream({
+    start(controller) {
+      c = controller;
+    }
+  });
+
+  const ws = new WritableStream({
+    write() { }
+  });
+
+  pipeline(rs,
+           makeTransformStream(),
+           makeTransformStream(),
+           makeTransformStream(),
+           makeTransformStream(),
+           ws,
+           common.mustCall((err) => {
+             assert.strictEqual(err?.message, 'kaboom');
+           }));
+
+  c.error(new Error('kaboom'));
+}
+
+{
+  const values = [];
+
+  const r = new Readable({
+    read() { }
+  });
+
+  const ws = new WritableStream({
+    write(chunk) {
+      values.push(chunk?.toString());
+    }
+  });
+
+  pipeline(r, ws, common.mustSucceed(() => {
+    assert.deepStrictEqual(values, ['helloworld']);
+  }));
+
+  r.push('hello');
+  r.push('world');
+  r.push(null);
+}
+
+{
+  const values = [];
+  let c;
+  const rs = new ReadableStream({
+    start(controller) {
+      c = controller;
+    }
+  });
+
+  const w = new Writable({
+    write(chunk, encoding, callback) {
+      values.push(chunk?.toString());
+      callback();
+    }
+  });
+
+  pipeline(rs, w, common.mustSucceed(() => {
+    assert.deepStrictEqual(values, ['hello', 'world']);
+  }));
+
+  c.enqueue('hello');
+  c.enqueue('world');
+  c.close();
+}
+
+{
+  const values = [];
+  let c;
+  const rs = new ReadableStream({
+    start(controller) {
+      c = controller;
+    }
+  });
+
+  const ws = new WritableStream({
+    write(chunk) {
+      values.push(chunk?.toString());
+    }
+  });
+
+  const t = new Transform({
+    transform(chunk, encoding, callback) {
+      callback(null, chunk?.toString().toUpperCase());
+    }
+  });
+
+  pipeline(rs, t, ws, common.mustSucceed(() => {
+    assert.deepStrictEqual(values, ['HELLOWORLD']);
```

Comment:
I have updated to not use pipeToWeb but that means the behaviour changes slightly is this correct? for context same behaviour as I described here https://github.com/nodejs/node/discussions/46381 is this correct behaviour? @ronag